src/consensus.rs

```diff
@@ -437,24 +442,23 @@ impl <L, M> Consensus<L, M> where L: Log, M: StateMachine {
                     self.persistent_log.entry(prev_log_index).unwrap().0
                 };
 
-            let mut message = MallocMessageBuilder::new_default();
-            {
-                let mut request = message.init_root::<message::Builder>()
-                                         .init_append_entries_request();
-                request.set_term(local_term.into());
-                request.set_prev_log_index(prev_log_index.into());
-                request.set_prev_log_term(prev_log_term.into());
-                request.set_leader_commit(self.commit_index.into());
-
-                let from_index = Into::<u64>::into(next_index);
-                let until_index = Into::<u64>::into(local_latest_log_index) + 1;
-                let mut entries = request.init_entries((until_index - from_index) as u32);
-                for (n, index) in (from_index..until_index).enumerate() {
-                    entries.set(n as u32, self.persistent_log.entry(LogIndex::from(index)).unwrap().1);
-                }
+            let from_index = Into::<u64>::into(next_index);
+            let until_index = Into::<u64>::into(local_latest_log_index) + 1;
+
+            let mut entries = vec![];
+            for index in (from_index .. until_index) {
+                entries.push(self.persistent_log.entry(LogIndex::from(index)).unwrap())
             }
+
+            let message = messages::append_entries_request(
+                local_term.into(),
+                prev_log_index.into(),
+                prev_log_term.into(),
+                &entries,
+                self.commit_index.into());
+
             self.leader_state.set_next_index(from, local_latest_log_index + 1);
-            actions.peer_messages.push((from, Rc::new(message)));
+            actions.peer_messages.push((from, message));
```

Comment:
the part of the diff above looks like it duplicates code with your change in `peer_connection_reset`. Seems worth refactoring.